src/odemis/util/spectrum.py

```diff
@@ -72,7 +72,7 @@ def get_wavelength_per_pixel(da):
             raise ValueError("Length of wavelength list does not match length of wavelength data.")
         return wl
     except KeyError:
-        logging.error("No MD_WL_LIST metadata available")
+        raise ValueError("No MD_WL_LIST metadata available")
```

Comment:
Please, change to KeyError, as it used to behave, and as it is described in the docstring.